### PR TITLE
Removes the Double Baton 'Tech' / Fixes targeting issues related to the stunbaton

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
## About The Pull Request
You now have to obey the cooldown of chain hitting as it was designed while still allowing for multi-user chainstunning.

Also fixes an issue with stun baton targeting.

## Why It's Good For The Game
It seems like (is) an oversight, as the cooldown's purpose is to avoid repeated attacks from the player without having to infringe on the stunner's next_move (IE so they can follow up with cuffs).

## Changelog
:cl: Cobby
balance: You will tend to miss attacks if trying to stun someone with a baton recently after using a different baton to stun them.
/:cl:

### CODE REVIEW

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -217,6 +217,9 @@
 	else
 		if(!deductcharge(cell_hit_cost))
 			return FALSE
+	if(HAS_TRAIT_FROM(L, TRAIT_IWASBATONED, user))
```
Comment (ATH1909) [inline code/game/objects/items/stunbaton.dm]: will this still deduct charge from the baton?

**Comment by ATH1909** [comment]
IMO, this shouldn't send a message back to the batonner or do a melee strike animation at all, since someone who's staminacritted dodging a baton strike doesn't make much sense. Move the check to before any of the animations and stuff, and make it silent like the current check.

**Comment by Tlaltecuhtli** [comment]
can you miss a corpse

**Comment by ExcessiveUseOfCobblestone** [comment]
if it doesn't provide any sort of feedback it's gonna look like a bug to the end user.

the "current check", assuming we're talking about the cooldown check, provides a message stating the baton is still charging.

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> can you miss a corpse

****trooper syndrome

**Comment by interestingusernam3** [comment]
IMO this should make the batonner hit themselves with the baton instead of making the target dodge.

**Review by spookydonut — APPROVED** [review]

**Comment by SinfulBliss** [comment]
B-but...He promised...
oranges batong

**Comment by RuneQuester** [comment]
He lied to you,as he has lied before

**Pull request merged**